api/src/test/scala/za/co/absa/loginsvc/rest/actuator/LdapHealthServiceTest.scala

```diff
@@ -0,0 +1,82 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.loginsvc.rest.actuator
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.springframework.boot.actuate.health.Health
+import org.springframework.boot.test.context.SpringBootTest
+import za.co.absa.loginsvc.rest.config.auth.{ActiveDirectoryLDAPConfig, LdapUserCredentialsConfig, ServiceAccountConfig, UsersConfig}
+import za.co.absa.loginsvc.rest.config.provider.AuthConfigProvider
+import za.co.absa.loginsvc.rest.service.actuator.LdapHealthService
+
+@SpringBootTest
+class LdapHealthServiceTest extends AnyFlatSpec with Matchers {
+
+  private val integratedCfg = LdapUserCredentialsConfig("svc-ldap", "password")
+  private val serviceAccountCfg = ServiceAccountConfig(
+    "CN=%s,OU=Users,OU=Accounts,DC=domain,DC=subdomain,DC=com",
+    Option(integratedCfg),
+    None)
+  private val ldapCfgZeroOrder = ActiveDirectoryLDAPConfig(
+    "some.domain.com",
+    "ldaps://some.domain.com:636/",
+    "SomeAccount",
+    0,
+    serviceAccountCfg,
+    None,
+    None,
+    None)
+
+  "LdapHealthService" should "Return Up on Order 0" in {
+    val configProvider = new AuthConfigProvider {
+      override def getLdapConfig: Option[ActiveDirectoryLDAPConfig] = Some(ldapCfgZeroOrder)
+
+      override def getUsersConfig: Option[UsersConfig] = None
+    }
+    val ldapHealthService: LdapHealthService = new LdapHealthService(configProvider)
+    val health = ldapHealthService.health()
+
+    health shouldBe Health.up().withDetail("reason", "ldap order parameter is set to 0. ldap is disabled.").build()
+  }
+
+  "LdapHealthService" should "Return Up when ActiveDirectoryLDAPConfig is None" in {
+    val configProvider = new AuthConfigProvider {
+      override def getLdapConfig: Option[ActiveDirectoryLDAPConfig] = None
+
+      override def getUsersConfig: Option[UsersConfig] = None
+    }
+    val ldapHealthService: LdapHealthService = new LdapHealthService(configProvider)
+    val health = ldapHealthService.health()
+
+    health shouldBe Health.up().withDetail("reason", "ldap authentication not found in configuration. ldap is disabled.").build()
+  }
+
+  "LdapHealthService" should "Return Down when Ldap connection fails" in {
+    val configProvider = new AuthConfigProvider {
+      override def getLdapConfig: Option[ActiveDirectoryLDAPConfig] = Some(ldapCfgZeroOrder.copy(order = 2))
+
+      override def getUsersConfig: Option[UsersConfig] = None
+    }
+    val ldapHealthService: LdapHealthService = new LdapHealthService(configProvider)
+    val health = ldapHealthService.health()
+
+    health shouldBe Health.down().withDetail("reason", "Failed to connect: some.domain.com:636").build()
+  }
```

Comment:
Do I understand correctly that this tests relies on the fact that connection to `some.domain.com:636` will fail?
Why not rather mock throwing the exception?

This approach may be useful (or even needed) if you decide to test an unexpected exception, too.